src/vos/vos_internal.h

```diff
@@ -228,6 +228,35 @@ struct vos_dtx_act_ent {
 	int				 dae_rec_cap;
 };
 
+extern struct vos_tls	*standalone_tls;
+#ifdef VOS_STANDALONE
+unsigned int tmp_count;
+#define VOS_TIME_START(start, op)		\
+do {						\
+	if (standalone_tls->vtl_dp == NULL)	\
+		break;				\
+	start = daos_get_ntime();		\
+} while (0)
+
+#define VOS_TIME_END(start, op)			\
+do {						\
+	struct daos_profile *dp;		\
+	int time_msec;				\
+						\
+	dp = standalone_tls->vtl_dp;		\
+	if ((dp) == NULL || start == 0)		\
+		break;				\
+	time_msec = (daos_get_ntime() - start)/1000; \
+	daos_profile_count(dp, op, time_msec);	\
+} while (0)
+
+#else
+
+#define VOS_TIME_START(start, op) D_TIME_START(start, op)
+#define VOS_TIME_END(start, op) D_TIME_END(start, op)
+
+#endif
```

Comment:
suggestion: looks dup with the code in iosrv, If you use pthread_get/setspecific() directly to get/set profile TLS, then above code could be moved into common lib?